toutv/dl.py

```diff
@@ -297,10 +302,11 @@ def download(self):
         self._logger.debug('parsed M3U8 file: {} total segments'.format(self._total_segments))
 
         # get decryption key
-        uri = self._segments[0].key.uri
-        r = self._do_request(uri)
-        self._key = r.content
-        self._logger.debug('decryption key: {}'.format(self._key))
+        if self._segments[0].key:
```

Comment:
Ok, I think it's a good idea to keep the possibility there being a key, in case it's just temporary.